collectors/logsmanagement.plugin/plugin_logsmanagement_web_log.c

```diff
@@ -0,0 +1,995 @@
+        chart_data->dim_req_method_propfind = rrddim_add(chart_data->st_req_methods, "PROPFIND", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_proppatch = rrddim_add(chart_data->st_req_methods, "PROPPATCH", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_put = rrddim_add(chart_data->st_req_methods, "PUT", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_rebind = rrddim_add(chart_data->st_req_methods, "REBIND", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_report = rrddim_add(chart_data->st_req_methods, "REPORT", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_search = rrddim_add(chart_data->st_req_methods, "SEARCH", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_trace = rrddim_add(chart_data->st_req_methods, "TRACE", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_unbind = rrddim_add(chart_data->st_req_methods, "UNBIND", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_uncheckout = rrddim_add(chart_data->st_req_methods, "UNCHECKOUT", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_unlink = rrddim_add(chart_data->st_req_methods, "UNLINK", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_unlock = rrddim_add(chart_data->st_req_methods, "UNLOCK", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_update = rrddim_add(chart_data->st_req_methods, "UPDATE", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_method_updateredirectref = rrddim_add(chart_data->st_req_methods, "UPDATEREDIRECTREF", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+    }
+
+    /* Request protocol - initialise */
+    if(p_file_info->parser_config->chart_config & CHART_REQ_PROTO){
+        chart_data->st_req_proto = rrdset_create_localhost(
+                (char *) p_file_info->chart_name
+                , "http versions"
+                , NULL
+                , "http versions"
+                , NULL
+                , "Requests Per HTTP Version"
+                , "requests/s"
+                , "logsmanagement.plugin"
+                , NULL
+                , NETDATA_CHART_PRIO_REQ_PROTO
+                , p_file_info->update_every
+                , RRDSET_TYPE_AREA
+        );
+        chart_data->dim_req_proto_http_1 = rrddim_add(chart_data->st_req_proto, "1.0", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_proto_http_1_1 = rrddim_add(chart_data->st_req_proto, "1.1", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_proto_http_2 = rrddim_add(chart_data->st_req_proto, "2.0", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_req_proto_other = rrddim_add(chart_data->st_req_proto, "other", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
```

Comment:
Instead `other` can we add two other dimensions (`higher`, `lower`) or `unknown`? In theory when we do not detect `http` version, we should create another PR to add it. :thinking:

Reply:
That's a good point, but I think "other" works for now. I followed the same approach the Go plugin takes, and it doesn't seem like we have any complains from users there.